bolts-tasks/src/main/java/bolts/UnobservedErrorNotifier.java

```diff
@@ -0,0 +1,29 @@
+package bolts;
+
+/**
+ * This class is used to retain a faulted task until either its error is observed or it is
+ * finalized. If it is finalized with a task, then the uncaught exception handler is exected
+ * with an UnobservedTaskException.
+ */
+public class UnobservedErrorNotifier {
+  private Task<?> task;
+
+  public UnobservedErrorNotifier(Task<?> task) {
+      this.task = task;
+  }
+
+  @Override
+  protected void finalize() throws Throwable {
+    Task faultedTask = this.task;
+    if (faultedTask != null) {
+      Task.UnobservedExceptionHandler ueh = Task.getUnobservedExceptionHandler();
+      if (ueh != null) {
+        ueh.unobservedException(faultedTask, new UnobservedTaskException(faultedTask.getError()));
+      }
+    }
+  }
```

Comment:
one small edit: we should be using `try { ... } finally { super.finalize() }` here